### PR TITLE
recipes_source/amp_recipe.rst 번역

### PR DESCRIPTION
**Pull request by hihunjin**
## 라이선스 동의
_변경해주시는 내용에 BSD 3항 라이선스가 적용됨을 동의해주셔야 합니다._<br />
_더 자세한 내용은 [기여하기 문서](https://github.com/9bow/PyTorch-tutorials-kr/blob/master/CONTRIBUTING.md)를 참고해주세요._<br />
_동의하시면 아래 `[ ]`를 `[x]`로 만들어주세요._<br />

- [x] 기여하기 문서를 확인하였으며, 본 PR 내용에 BSD 3항 라이선스가 적용됨에 동의합니다.

## 관련 이슈 번호
_이 Pull Request와 관련있는 이슈 번호를 적어주세요._<br />
_이슈 또는 PR 번호 앞에 #을 붙이시면 제목을 바로 확인하실 수 있습니다. (예. #999 )_

- **이슈 번호**: #214 

## PR 종류
_이 PR에 해당되는 종류 앞의 `[ ]`을 `[x]`로 변경해주세요._<br />
- [ ] 오탈자를 수정하거나 번역을 개선하는 기여
- [x] 번역되지 않은 튜토리얼을 번역하는 기여
- [ ] 공식 튜토리얼 내용을 반영하는 기여
- [ ] 위 종류에 포함되지 않는 기여

## PR 설명
_이 PR로 무엇이 달라지는지 대략적으로 알려주세요._

[https://pytorch.org/tutorials/recipes/recipes/amp_recipe.html#automatic-mixed-precision](https://pytorch.org/tutorials/recipes/recipes/amp_recipe.html#automatic-mixed-precision)
번역

### CODE REVIEW

**Comment by 9bow** [comment]
감사합니다. 빠른 시일 내에 리뷰하도록 하겠습니다. :)

File: recipes_source/amp_recipe.rst [hunk 1]
```diff
@@ -0,0 +1,248 @@
+AUTOMATIC MIXED PRECISION(자동 혼합 정밀도)
+===========================================================
+
+**Author**: `Michael Carilli <https://github.com/mcarilli>`_
+  **번역**: `하헌진 <https://github.com/hihunjin>`_
+
+
+`torch.cuda.amp`_ 는 mixed precision을 사용할 수 있는 편리한 메서드를 제공합니다. precision은 ``torch.float32`` (``float``)과, ``torch.float16`` (``half``)를 사용하는 operations들이 있습니다. Linear 층과 convolusion 층들과 같은 operations들은, ``float16`` 에서 더욱 빠릅니다. reduction과 같은 operations들은 ``float32`` 의 동적인 범주 계산을 요구하기도 합니다. Mixed precision은 각각의 operation에 맞는 적절한 데이터 타입을 맞춰줍니다. 따라서 네트워크의 실행 시간과 메모리 할당량을 줄일 수 있습니다.
```
Comment (9bow) [inline recipes_source/amp_recipe.rst:8]: reST 문법 상 \` 뒤에 한글이 오는 경우에는 공백 또는 `\ `이 필요하니 수정 부탁드립니다. (예. ``float``과 => ``float``\ 과 )

용어들을 어디까지 번역을 해야하는지에 대해서는 번역을 하면서도 계속 헷갈리기도 하고, 사실 리뷰할 때도 가장 힘든 부분이긴 합니다 ^^; 
번역 기여자분께서 선호하시는 스타일로 가급적 진행하려고 하지만, 통일성을 위해 한국어로 번역이 가능한 경우에는 한국어로 번역 및 추가적인 설명이 필요한 경우에는 원문을 괄호 안에 병기해주시면 어떨까 싶습니다.
(예를 들어, 기존에 Linear layer와 Convolution layer는 각각 선형 (계)층과 합성곱 (계)층으로 번역이 되었으며, operation은 연산과 같이 번역하였습니다. 비슷한 관점에서 network는 신경망으로 번역해주시는 것이 조금 더 편안하지 않을까 싶습니다.)
그 외, 판단하시기 애매하신 경우에는 이미 번역되어 널리 읽히고 있는 문서들(예. [Tensorflow 번역 문서](https://www.tensorflow.org/guide/mixed_precision)) 등을 참고해주시면 어떨까 싶습니다. :)

추가적으로 내용 상으로는 Automatic Mixed Precision은 Automatic Mixed Precision (training)으로 '자동화된 혼합 정밀도' 정도가 어떨까요?

File: recipes_source/amp_recipe.rst [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+보통, "automatic mixed precision training"은 `torch.cuda.amp.autocast`_ 와 `torch.cuda.amp.GradScaler`_ 를 함께 사용합니다.
+
+이 레시피는 먼저 디폴트 precision에서 간단한 네트워크의 성능을 측정하고, 다음으로 ``autocast`` 와 ``GradScaler`` 를 추가하여, 같은 네트워크에서 향상된 성능을 살펴봅니다.
```
Comment (9bow) [inline recipes_source/amp_recipe.rst:12]: 디폴트는 기본으로 바꿔주시면 조금 더 편안하지 않을까 싶습니다. ^^

File: recipes_source/amp_recipe.rst [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            loss = loss_fn(output, target)
+            loss.backward()
+            opt.step()
+            opt.zero_grad() # set_to_none=True 를 사용하면 속도를 약간 향상할 수 있습니다.
+    end_timer_and_print("Default precision:")
+
+autocast 사용하기
+---------------------------
+
+`torch.cuda.amp.autocast`_ 의 인스턴스는 컨텍스트 매니저로써 mixed precision을 수행하는 코드 리전을 제공합니다.
```
Comment (9bow) [inline recipes_source/amp_recipe.rst:97]: 원문에서 region은 with 구문으로 발생하는 ContextManager가 제공하는 블럭을 뜻하는 것처럼 보이는데요,
> Instances of `torch.cuda.amp.autocast <https://pytorch.org/docs/stable/amp.html#autocasting>`_
> serve as context managers that allow regions of your script to run in mixed precision.
`torch.cuda.amp.autocast 인스턴스는 컨텍스트 매니저로 제공되어, 혼합 정밀도로 실행되는 코드 영역(region)을 제공합니다` 등과 같은 번역은 어떠실까요?

**Review by 9bow — CHANGES_REQUESTED** [review]
안녕하세요, @hihunjin 님. PR 감사합니다.
먼저 보내주신 PR은 현재 존재하지 않는 문서를 새로 만들어주셨는데요, 
AMP recipe 파일은 `recipes_source/amp_recipe.rst`에 위치해 있지 않으며, 아래 위치에 존재하고 있습니다.
https://github.com/9bow/PyTorch-tutorials-kr/blob/master/recipes_source/recipes/amp_recipe.py

해당 문서를 기준으로 다시 PR 진행 부탁드리며, 문서 전반적으로 외부 링크가 생략되어 있습니다.
(예. PR 보내주신 문서 10번째 라인의 `torch.cuda.amp.autocast`_ 와 `torch.cuda.amp.GradScaler`_ 는  각각 `torch.cuda.amp.autocast <https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.autocast>`_ 와
`torch.cuda.amp.GradScaler <https://pytorch.org/docs/stable/amp.html#torch.cuda.amp.GradScaler>`_ 가 되어야 합니다.)

그 외 추가적으로 살펴봐주셨으면 하는 부분들을 간략하게 덧글로 달았으니 함께 확인하시고 다시 PR 열어주시기를 부탁드립니다.

**Comment by 9bow** [comment]
위 말씀드린 것처럼 원문을 기준으로 다시 PR 보내주시기를 부탁드립니다. 감사합니다.

File: recipes_source/amp_recipe.rst (hunk 1 above)
Comment (hihunjin) [reply recipes_source/amp_recipe.rst:8]: Automatic Mixed Precision : 자동 혼합 정밀도
로 번역하겠습니다.

**Comment by hihunjin** [comment]
@9bow 혹시 ReLU는 어떻게 번역하는게 좋을까요? 공유해주신 Tensorflow 문서와 pytorch tutorial kr문서를 봐도 찾기 어려워서 질문을 남깁니다.

**Comment by hihunjin** [comment]
@9bow Automatic Mixed Precision을 AMP로 줄여서 말하기도 하는데요, 이는 어떻게 표현하는게 좋을까요? 
1) AMP
2) 자동 혼합 정밀도

**Comment by 9bow** [comment]
네, 제가 번역했던 문서들 중에서는 SGD가 비슷한 사례일 것 같은데요, 보통은 `한국어(영어)` 식으로 원문을 표기했었는데 해당 경우에는 `확률적 경사하강법(SGD; Stochastic Gradient Descent)`과 같이 번역을 했었습니다. ([참고 링크](https://github.com/9bow/PyTorch-tutorials-kr/blob/master/beginner_source/basics/optimization_tutorial.py#L128)입니다.)

개인적으로는 레시피와 초급(각 `recipes_source`와 `beginner_source` 디렉토리 내의) 문서들은 가급적 번역된 문서만을 읽었을 때도 어느 정도 이해가 되면 좋을 것 같아서 약간의 의역을 포함하여 작성하고 있으니 참고 부탁드립니다. (예를 들어, 부호화/encoding, 직렬화/serialization 과 같은 용어들은 이미 익숙하신 분들도 이해가 어려우실 수 있어 가급적 동일 문서 내에서라도 여러번 병기를 하려고 하고 있습니다.)

감사합니다. :)
